CHANGELOG.md

```diff
@@ -1,5 +1,62 @@
 # go-ipfs changelog
 
+Ipfs 0.4.16 is a fairly small release in terms of changes to the ipfs codebase,
+but it contains a huge amount of changes and improvements from libraries we
+depend on, notably libp2p. This version of ipfs contains the changes for libp2p
+v6.0.5, In that release, we made significant changes to the codebase to allow
+for more easy integration of future transports and modules. We also improved
+some of our dialing logic, fixed a couple issues in the DHT, and added
+support for a new way to write ipfs multiaddrs.  The transport refactor now
+allows us to much more easily add in support for running libp2p (and by
+extension, ipfs) over QUIC, or using TLS instead of secio for encrypting
+connections. Our [QUIC
+transport](https://github.com/libp2p/go-libp2p-quic-transport) currently
+works, and can be plugged into libp2p manually (though note that it is
+still experimental, as the upstream spec is still in flux). Further work is
+needed to make enabling this inside ipfs easy and not require
+recompilation.  For more information on the refactor and libp2p v6.0.0, see
+the [release blog post](https://ipfs.io/blog/39-go-libp2p-6-0-0/).  Between
+libp2p v6.0.0 and v6.0.5, there have been some changes worth pointing out
+as well. The swarm dialer has been improved to have shortened timeouts when
+dialing peers in 'local' subnets. This prevents large dial timeouts from
+wasting dial time when the subnet you are dialing is not accessible. The
+TCP handshake timeout has also been dropped to 5 seconds, improving the
+performance of dials to non-existent addresses. In the DHT, we have fixed
+the query code to put records to the K closest peers we can actually
+connect to, which is a strict superset of the peers we previously put
+records to: The peers we could connect to out of the K closest we learned
+about.  Finally, we are changing the way that people write 'ipfs'
+multiaddrs.  Currently, ipfs multiaddrs look something like
+`/ip4/104.131.131.82/tcp/4001/ipfs/QmaCpDMGvV2BGHeYERUEnRQAwe3N8SzbUtfsmvsqQLuvuJ`.
+But caling them 'ipfs' multiaddrs is a bit misleading, as this is actually
+the multiaddr of a libp2p peer that runs ipfs. Other protocols using libp2p
+right now still have to use multiaddrs that say 'ipfs', without actually
+having anything to do with ipfs. Towards that, we are renaming them to
+'p2p' multiaddrs. Moving forward these addresses will be written as:
+`/ip4/104.131.131.82/tcp/4001/p2p/QmaCpDMGvV2BGHeYERUEnRQAwe3N8SzbUtfsmvsqQLuvuJ`.
+This release adds support for *parsing* both types of addresses into the
+same binary format, and the binary format is remaining exactly the same. A
+future release will have the ipfs daemon switch to *printing* out addresses
+this way, once a large enough portion of the network has upgraded.
```

Comment:
Whoa, this paragraph is huuuuge. Maybe a bulleted list would make it a bit easier to parse?

Reply:
Turned into paragraphs.